proto/minder/v1/minder.proto

```diff
@@ -3411,3 +3412,144 @@ message UpstreamEntityRef {
     // be stored in Minder.
     google.protobuf.Struct properties = 3;
 }
+
+// DataSource is a Data source instance. Data sources represent
+// external integrations that enrich the data in Minder, but do not
+// provide entities (that is, they're not providers).
+message DataSource {
+    // version is the version of the data source API.
+    string version = 1 [
+        (buf.validate.field).string = {
+            pattern: "^v\\d$",
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // type is the data source type
+    string type = 2 [
+        (buf.validate.field).string = {
+            pattern: "^data-source$",
+            max_len: 12,
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // context is the context in which the data source is evaluated.
+    // Note that in this case we only need the project in the
+    // context, since data sources are not provider-specific.
+    ContextV2 context = 3;
+
+    // name is the name of the data source.
+    string name = 4 [
+        (buf.validate.field).string = {
+            pattern: "^[A-Za-z][-[:word:]]*$",
+            max_len: 200,
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // id is the unique identifier of the data source.
+    string id = 5 [
+        (buf.validate.field).string = {uuid: true},
+        (google.api.field_behavior) = OUTPUT_ONLY
+    ];
+
+    // input is the input configuration for the data source.
+    // It allows us to configure a JSON schema for the input data.
+    google.protobuf.Struct input = 6;
+
+    oneof driver {
+        // rest is the REST data source driver.
+        RestDataSource rest = 7;
+    }
+}
+
+// RestDataSource is the REST data source driver.
+message RestDataSource {
+    // endpoint is the URL of the REST API.
+    string endpoint = 1 [
+        (buf.validate.field).string = {
+            pattern: "^https?://.*$",
+            max_len: 800,
+        }
+    ];
+
+    // method is the HTTP method to use for the request.
+    string method = 2 [
+        (buf.validate.field).string = {
+            in: ["GET", "POST", "PUT", "PATCH", "DELETE"]
+        },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
+
+    // headers is a map of headers to send with the request.
+    google.protobuf.Struct headers = 3;
```

Comment:
Should this be a `map[string]string`, rather than any type of JSON object?

Are any of the following valid?

```yaml
headers: help
```

```yaml
headers:
- one
- two
```

```yaml
headers:
  accept:
    json: ["text", "xml"]
```

Reply:
Good catch!